src/components/ImageCard/HotspotContent.jsx

```diff
@@ -0,0 +1,166 @@
+    value: PropTypes.any,
+    icon: PropTypes.string,
+    color: PropTypes.string,
+  }),
+  /** the locale to use for formatting numeric values */
+  locale: PropTypes.string,
+  /** ability to render icon by name */
+  renderIconByName: PropTypes.func,
+};
+
+const defaultProps = {
+  title: null,
+  description: null,
+  values: {},
+  attributes: [],
+  hotspotThreshold: null,
+  renderIconByName: null,
+  locale: 'en',
+};
+
+const HotspotContent = ({
+  title,
+  description,
+  attributes,
+  values,
+  hotspotThreshold,
+  locale,
+  renderIconByName,
+}) => (
+  <div className={`${iotPrefix}--hotspot-content`}>
+    {typeof title === 'string' ? (
+      <h4 title={title}>{title}</h4>
+    ) : React.isValidElement(title) ? (
+      title
+    ) : null}
+    {description && <p>{description}</p>}
+    {attributes.map(({ thresholds, dataSourceId, label, unit, precision }) => {
+      // look for attribute specific thresholds first
+      let attributeThresholdMatch = null;
+      if (!isEmpty(thresholds)) {
+        const matchingAttributeThresholds = findMatchingThresholds(
+          thresholds.map(threshold => ({ ...threshold, dataSourceId })),
+          values,
+          dataSourceId
+        );
+        if (matchingAttributeThresholds && !isEmpty(matchingAttributeThresholds)) {
+          [attributeThresholdMatch] = matchingAttributeThresholds;
+        }
+      }
+      const thresholdMatch =
+        attributeThresholdMatch ||
+        (hotspotThreshold && hotspotThreshold.dataSourceId === dataSourceId // then see if the parent threshold might match this attribute
+          ? hotspotThreshold
+          : null);
+      const value = isNil(values[dataSourceId]) ? '--' : values[dataSourceId];
+      const thresholdIcon =
+        thresholdMatch && thresholdMatch.dataSourceId === dataSourceId && thresholdMatch.icon ? (
+          <CardIcon
+            icon={thresholdMatch.icon}
+            color={thresholdMatch.color}
+            title={`${thresholdMatch.dataSourceId} ${thresholdMatch.comparison} ${
+              typeof thresholdMatch.value === 'number'
+                ? formatNumberWithPrecision(thresholdMatch.value, null, locale)
+                : thresholdMatch.value
+            }`}
+            width={16}
+            height={16}
+            renderIconByName={renderIconByName}
+          />
+        ) : null;
+      return (
+        <div key={`attribute-${dataSourceId}`} style={{ display: 'flex', alignItems: 'center' }}>
+          <div style={{ flex: 1, paddingRight: 20 }}>
+            <span style={{ fontWeight: 'bold' }}>{label}:</span>
+          </div>
+          <div
+            style={{
+              flex: 0,
+              whiteSpace: 'nowrap',
+              display: 'flex',
+              alignItems: 'center',
+            }}
+          >
+            {thresholdIcon}
+            <span
+              style={{
+                paddingLeft: thresholdIcon ? 5 : 0,
```

Comment:
this logic is to be performed to populate a css custom property that can then be used in your sass files